src/c_api/c_api.cc

```diff
@@ -604,20 +604,58 @@ XGB_DLL int XGBoosterPredict(BoosterHandle handle,
   CHECK_HANDLE();
   auto *learner = static_cast<Learner*>(handle);
   auto& entry = learner->GetThreadLocal().prediction_entry;
-  learner->Predict(
-      *static_cast<std::shared_ptr<DMatrix>*>(dmat),
-      (option_mask & 1) != 0,
-      &entry.predictions, ntree_limit,
-      static_cast<bool>(training),
-      (option_mask & 2) != 0,
-      (option_mask & 4) != 0,
-      (option_mask & 8) != 0,
-      (option_mask & 16) != 0);
+  ntree_limit = GetLayerTrees(ntree_limit, learner);
```

Comment:
I'm confused about variable names - this is still ntree_limit, wouldn't this now be called layer_end?

Reply:
That's the old API, I don't want to break it for 1.4.   I will rename the variable.